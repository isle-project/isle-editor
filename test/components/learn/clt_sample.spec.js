// MODULES //

import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import LearnSampleCLT from '@isle-project/components/learn/clt-sample/main.js';
import heartdisease from './../fixtures/heartdisease.json';


// TESTS //

describe( '<LearnSampleCLT />', function test() {
	it( 'renders a learning component', () => {
		const learning = renderer.create( <LearnSampleCLT
			data={heartdisease}
			variables={[ 'Gender', 'Drugs', 'Complications', 'Cost', 'Age', 'Interventions', 'ERVisit', 'Comorbidities', 'Duration' ]}
		/> );
		expect( learning ).toMatchSnapshot();
	});
});
