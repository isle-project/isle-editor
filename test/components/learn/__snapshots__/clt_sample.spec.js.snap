// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LearnSampleCLT /> renders a learning component 1`] = `
<div
  className="container"
  style={
    Object {
      "maxWidth": 1200,
    }
  }
>
  <div
    className="row"
  >
    <div
      className="card"
      style={
        Object {
          "margin": "2%",
          "width": "96%",
        }
      }
    >
      <div
        className="card-body"
      >
        <div
          className="row"
        >
          <div
            className="col-md-6"
          >
            <form
              className="input"
              onSubmit={[Function]}
              style={Object {}}
            >
              <div>
                <label
                  htmlFor="select-input-1"
                >
                  select-variable
                </label>
                <div
                  className=" css-b62m3t-container"
                  id="select-input-1"
                  onKeyDown={[Function]}
                >
                  <span
                    className="css-1f43avz-a11yText-A11yText"
                    id="react-select-2-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    className="css-1f43avz-a11yText-A11yText"
                  />
                  <div
                    className=" css-b82su3-control"
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                  >
                    <div
                      className=" css-319lph-ValueContainer"
                    >
                      <div
                        className=" css-bqaabt-placeholder"
                        id="react-select-2-placeholder"
                      >
                        select-placeholder
                      </div>
                      <div
                        className=" css-6j8wv5-Input"
                        data-value=""
                      >
                        <input
                          aria-autocomplete="list"
                          aria-controls="react-select-2-listbox"
                          aria-describedby="react-select-2-placeholder"
                          aria-expanded={false}
                          aria-haspopup={true}
                          aria-label="select-between Gender, Drugs, Complications, Cost, Age, Interventions, ERVisit, Comorbidities, Duration"
                          aria-owns="react-select-2-listbox"
                          autoCapitalize="none"
                          autoComplete="off"
                          autoCorrect="off"
                          className=""
                          disabled={false}
                          id="react-select-2-input"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          role="combobox"
                          spellCheck="false"
                          style={
                            Object {
                              "background": 0,
                              "border": 0,
                              "color": "inherit",
                              "font": "inherit",
                              "gridArea": "1 / 2",
                              "label": "input",
                              "margin": 0,
                              "minWidth": "2px",
                              "opacity": 1,
                              "outline": 0,
                              "padding": 0,
                              "width": "100%",
                            }
                          }
                          tabIndex={0}
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      className=" css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        className=" css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        className=" css-tlfecz-indicatorContainer"
                        onMouseDown={[Function]}
                        onTouchEnd={[Function]}
                      >
                        <svg
                          aria-hidden="true"
                          className="css-tj5bde-Svg"
                          focusable="false"
                          height={20}
                          viewBox="0 0 20 20"
                          width={20}
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <input
                    name="form-field-name"
                    type="hidden"
                    value=""
                  />
                </div>
              </div>
            </form>
          </div>
          <div
            className="col-md-6"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="fade alert alert-info show"
    role="alert"
    style={
      Object {
        "fontSize": 24,
      }
    }
  >
    require-numeric-or-binary
  </div>
</div>
`;
